src/chunk.c

```diff
@@ -2194,34 +2203,130 @@ ts_chunk_show_chunks(PG_FUNCTION_ARGS)
 		else
 			time_type = InvalidOid;
 
+		/* note that arg_types will be the same for all specified "ANY" elements for a given call */
+		arg_type = InvalidOid;
 		if (!PG_ARGISNULL(1))
-			older_than = ts_time_value_from_arg(PG_GETARG_DATUM(1),
-												get_fn_expr_argtype(fcinfo->flinfo, 1),
-												time_type);
+		{
+			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 1);
+			older_than = ts_time_value_from_arg(PG_GETARG_DATUM(1), arg_type, time_type, true);
+			older_newer = true;
+		}
 
 		if (!PG_ARGISNULL(2))
-			newer_than = ts_time_value_from_arg(PG_GETARG_DATUM(2),
-												get_fn_expr_argtype(fcinfo->flinfo, 2),
-												time_type);
+		{
+			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 2);
+			newer_than = ts_time_value_from_arg(PG_GETARG_DATUM(2), arg_type, time_type, true);
+			older_newer = true;
+		}
+
+		/*
+		 * We cannot have a mix of [older_than/newer_than] and [created_before/created_after].
+		 * So, check that first.
+		 */
+
+		if (!PG_ARGISNULL(3))
+		{
+			if (older_newer)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("cannot specify \"older_than\" or \"newer_than\" together with "
+								"\"created_before\""
+								"or \"created_after\""),
+						 errhint("\"older_than\" and/or \"newer_than\" is recommended with "
+								 "\"time\"-like partitioning"
+								 " and  \"created_before\" and/or \"created_after\" is recommended "
+								 "with \"integer\"-like"
+								 " partitioning.")));
```

Comment:
Not convinced we need this recommendation here. We don't know exactly how users will want to use this functionality. Perhaps there are use cases where creation time makes sense even with time-like partitioning. I would just avoid adding anything that locks down certain usage.


If necessary, we can explain how we believe one can use this in docs.

Reply:
Agreed. Will remove the HINT part which locks down certain usages.